### PR TITLE
Cce bugfix bundle

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Fixes a handful of hard-to-find bugs and reasoning mistakes that I've run across in practical usage of the CCE system.

### Types of changes:

- [x] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Evolution/Systems/Cce/ScriPlusValues.cpp [hunk 1]
```diff
@@ -254,10 +254,11 @@ void CalculateScriPlusValue<Tags::ScriPlus<Tags::Psi2>>::apply(
       -0.5 * get(boundary_r) *
       (-exp_2_beta_at_scri * (conj(ethbar_dy_q_at_scri) +
                               eth_r_divided_by_r_view * conj(dy_q_at_scri)) +
-       get(boundary_r) * (ethbar_dy_dy_u_at_scri +
-                          conj(eth_r_divided_by_r_view) * dy_dy_u_at_scri +
-                          conj(ethbar_dy_dy_u_at_scri) +
-                          eth_r_divided_by_r_view * conj(dy_dy_u_at_scri)) +
+       get(boundary_r) *
+           (ethbar_dy_dy_u_at_scri +
+            2.0 * conj(eth_r_divided_by_r_view) * dy_dy_u_at_scri +
```
Comment (kidder) [inline src/Evolution/Systems/Cce/ScriPlusValues.cpp:259]: is the documentation correct ?

File: src/Evolution/Systems/Cce/ScriPlusInterpolationManager.hpp [hunk 2]
```diff
@@ -482,6 +492,12 @@ struct ScriPlusInterpolationManager<VectorTypeToInterpolate, Tags::Du<Tag>> {
     return argument_interpolation_manager_.first_time_is_ready_to_interpolate();
   }
 
+  const std::deque<std::pair<double, double>>& get_u_bondi_ranges() const
+      noexcept {
+    return argument_interpolation_manager_.get_u_bondi_ranges();
+  }
+
+
```
Comment (kidder) [inline src/Evolution/Systems/Cce/ScriPlusInterpolationManager.hpp]: too many blank lines

File: src/NumericalAlgorithms/Spectral/SwshDerivatives.hpp [hunk 3]
```diff
@@ -225,12 +225,10 @@ struct AngularDerivativesImpl<tmpl::list<DerivativeTags...>,
   template <ComplexRepresentation FriendRepresentation,
             typename... DerivativeKinds, typename... ArgumentTypes,
             size_t... Is>
-  // NOLINTNEXTLINE(readability-redundant-declaratioon)
-  friend void angular_derivatives_impl(const std::tuple<ArgumentTypes...>&,
-                                       size_t, size_t,
-                                       std::index_sequence<Is...>,
-                                       tmpl::list<DerivativeKinds...>,
-                                       std::bool_constant<true>) noexcept;
+  friend void angular_derivatives_impl(  // NOLINT
```
Comment (kidder) [inline src/NumericalAlgorithms/Spectral/SwshDerivatives.hpp]: same question about NOLINT and redundant declaration

File: src/Utilities/ConstantExpressions.hpp [hunk 4]
```diff
@@ -51,7 +51,7 @@ SPECTRE_ALWAYS_INLINE constexpr size_t get_nth_bit(const size_t i,
 /// \ingroup ConstantExpressionsGroup
 /// \brief Compute the square of `x`
 template <typename T>
-SPECTRE_ALWAYS_INLINE constexpr decltype(auto) square(const T& x) {
+SPECTRE_ALWAYS_INLINE constexpr decltype(auto) square(const T& x) {  // NOLINT
```
Comment (kidder) [inline src/Utilities/ConstantExpressions.hpp]: what is NOLINTed here?

**Review by kidder — CHANGES_REQUESTED** [review]
you can squash immediately

File: src/Evolution/Systems/Cce/ScriPlusValues.cpp (hunk 1 above)
Comment (moxcodes) [reply src/Evolution/Systems/Cce/ScriPlusValues.cpp:259]: Yeah, the documentation leaves the Jacobian factors implicit, so doesn't need to change based on this realization.

File: src/Utilities/ConstantExpressions.hpp (hunk 4 above)
Comment (moxcodes) [reply src/Utilities/ConstantExpressions.hpp]: This one was `const-return-type` though now I think it's globally disabled so I'll just remove this `NOLINT`

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**